Add simple UI for voting poll metagraph (#17)
Constellation-Labs/metagraph-examples: examples/voting-poll/sample-ui/README.md

# Sample UI Project - Voting App

## Description

> This is a demo application built using Next.js 14 and npm. The application allows users to create polls and vote on them using their Stargazer wallet. Users can only vote if they have balance in their wallet.

Reviewer: ```suggestion
This is a demo web app frontend built using Next.js 14 and npm. The application allows users to create polls and vote on them using their Stargazer wallet. The metagraph is used as a backend for this web app.

Users use the L0 token balance in their Stargazer wallet as voting points for the polls. Each user  gets as many votes as they have tokens in their wallet. 
```

> This is a demo application built using Next.js 14 and npm. The application allows users to create polls and vote on them using their Stargazer DAG wallet. Users can only vote if they have balance in their wallet.

Reviewer: ```suggestion
This is a demo application built using Next.js 14 and npm. The application allows users to create polls and vote on them using their Stargazer wallet. Users can only vote if they have balance in their wallet.
```

We should just call it Stargazer wallet. It's actually being used as an L0 token wallet in this case.

## Features

+ Create new polls
+ Vote on existing polls
+ Authentication via Stargazer DAG wallet
+ Check wallet balance before voting
  
## Prerequisites

+ Node.js (v18 or higher)
+ npm (v9 or higher)

> + Stargazer DAG wallet

Reviewer: ```suggestion
+ Stargazer wallet
```

## Usage

### Development server

1. Install dependencies:

```sh
npm install
```

2. start the development server:

```sh
npm run dev
```

> Open http://localhost:5431 with your browser to see the result.

Reviewer: Note, ensure that your metagraph is running using `hydra status` before opening this app. It will fail if the metagraph is offline. 

(would be nice to have an error that says "cannot reach metagraph, check if it's online with hydra status" instead of the cannot fetch error)

## Stargazer wallet dependency

### Wallet balance
Users need to have a balance in their Stargazer wallet to vote. The application checks the wallet balance before allowing a vote.

> 

Reviewer: #### Adding token to Stargazer
In order to see metagraph tokens in your Stargazer wallet, you will need to add the metagraph as a custom Constellation token in your wallet. 

Open Stargazer wallet in Chrome, scroll to the bottom and click "manage tokens". Click the "+" in the upper right of the screen to add a custom token. 

Make sure your metagraph is started with `hydra start-genesis` and note the endpoint details output to your screen. You can also find these details using `hydra status`. 

Use the following details from your metagraph output:
- Network: Constellation
- L1 endpoint: metagraph-node-1 Currency L1 (http://localhost:9300)
- L0 endpoint: metagraph-node-1 Metagraph L0 (http://localhost:9200) 
- Metagraph ID: metagraph ID from your output
- Token name: Anything you want
- Token symbol: Anything you want.

#### Adding balance via genesis file

> To set up a wallet with balance, you can modify the genesis file on euclid-development-environment located at `euclid-development-environment/source/metagraph-l0/genesis/genesis.csv`. 

Reviewer: ```suggestion
To set up a wallet with balance, you can modify the genesis file in your local Euclid project located at `source/metagraph-l0/genesis/genesis.csv`.
```

> The file is a csv file, first column indicates the target address for funds, the second column indicates how much vp to allocate for the wallet (remember this value is raw value, so the actual value would be (rawValue/1e8)).

Reviewer: ```suggestion
to add your wallet address and the token balance it should have at metagraph genesis. To see the balance updated, you must rerun your metagraph with `hydra start-genesis`. 
The file is a csv file, first column indicates the target address for funds, the second column indicates the token balance to allocate for the wallet. This balance is denominated in datum, so the value added to the file should 1e8 times larger than number of tokens you want to create (e.g. 55 tokens would be 5500000000). 
```

> The file is a csv file, first column indicates the target address for funds, the second column indicates how much vp to allocate for the wallet (remember this value is raw value, so the actual value would be `(rawValue/1e8)`).

Reviewer: rawValue = datum

> Re-build tessellation images to see the reflected balance on the network.

Reviewer: ```suggestion
Restart your metagraph from genesis to see the balance reflected in your wallet: `hydra start-genesis`. 
```

> The file is a csv file, first column indicates the target address for funds, the second column indicates how much vp to allocate for the wallet (remember this value is raw value, so the actual value would be `(rawValue = datum)`).

Reviewer: ```suggestion
The file is a csv file, first column indicates the target address for funds, the second column indicates the token balance to allocate for the wallet. This balance is denominated in datum, so the value added to the file should 1e8 times larger than number of tokens you want to create (e.g. 55 tokens would be 5500000000). 
```
